cobc/pplex.l

```diff
@@ -2337,6 +2430,17 @@ start:
 			memset (buff, '\n', newline_count);
 			gotcr += newline_count;
 		}
+		if (in_area_a) {
+			/* (*area-a*): prepend special marker `#', indicating
+			   the current non-continuation line starts in area A
+			   (see "AREA_A" and "MAYBE_AREA_A"... in lexer rules
+			   above).
+
+			   Assumes `buff' is large enough for one extra char. */
+			memmove (buff + 1, buff, gotcr + 1);
+			buff[1] = '#';
```

Comment:
That is fishy. We move the buffer one byte to the right, then overwrite the first of the moved data by `#`.
This way we effectively insert `#` into the second position. That won't work if we have any code on the first column (like in terminal format).

I also _think_ that this code must be moved _before_ we add the \n in front of the buffer- otherwise the area-a token is not before the code it marks but after the previous code (which makes a difference for where some line numbers go).

I've adjusted both and the testsuite still passes (and my additional tests, too).
@nberth: Do you remember if this may raise issues we miss testcases for? Otherwise I'll commit these changes in about 10 hours.

Reply:
Yes that was actually designed at a time when we only had FIXED/COBOL85 format for which the area A check was relevant, and in these cases these issues don't apply.
Issues I can think of and that might not have been tested are about:
- first program line starting in area A (in terminal format indeed — unlike most tests in the testsuite);
- pseudo-text in which the first token starts in area A: these cases are always tricky to handle.